cedar-policy-core/src/entities.rs

```diff
@@ -185,6 +182,28 @@ impl Entities {
                 compute_tc(&mut entity_map, true).map_err(Box::new)?;
             }
         }
+        // Now that TC has been enforced, we can check action entities for
+        // conformance with the schema and add action entities to the store.
+        // This is fine to do after TC because the action hierarchy in the
+        // schema already satisfies TC, and action and non-action entities
+        // will never be in the same hierarchy.
```

Comment:
nit: although actions are not allowed to have non-action parents, I believe that (due to oversight) non-actions are allowed to have action parents.  If so, the comment is inaccurate, but the code may still be correct

Reply:
Schemas do not allow specifying "Action" in the "memberOfTypes" list (a by-product of #304), so by this point in the code we've ruled out any cases of non-actions with action parents.

Reply:
(Updated comment to clarify)